src/components/wallet/card/wallet-card.tsx

```diff
@@ -0,0 +1,220 @@
+    const fiat_wallets = ['aud', 'eur', 'gbp', 'usd'];
+    const wallet_title = wallet_card_data[wallet_name]?.display_name || wallet_name;
+    const wallet_currency = currency || wallet_card_data[wallet_name]?.default_currency;
+    const div_ref = useRef<HTMLDivElement>(null);
+
+    const updateBackground = () => {
+        // here we set background and pattern colors of svg background:
+        const default_primary_color = dark ? '#303030' : '#CFCFCF';
+        const default_secondary_color = dark ? '#323738' : '#d6dadb';
+        const custom_primary_color = demo ? '#FF6444' : wallet_card_data[wallet_name]?.colors.primary;
+        const custom_secondary_color = demo ? '#FF444F' : wallet_card_data[wallet_name]?.colors.secondary;
+        const svg = div_ref.current?.querySelector('svg');
+        if (svg) {
+            svg.querySelectorAll('path')[0].setAttribute('fill', dark ? '#151717' : '#fff');
+            if (!demo && wallet_name !== 'demo' && size !== 'small') {
+                svg.querySelectorAll('path')[1]?.setAttribute('stroke', dark ? '#fff' : '#0E0E0E');
+            }
+            svg.querySelectorAll('circle').forEach((circle: SVGCircleElement, index: number) => {
+                // For large & medium size background: Primary color paints the 1st circle in the bottom-right corner & the 3rd circle in the top-left corner,
+                // and Secondary color paints the 2nd circle in the top-right corner,
+                // For small size background: Primary color paints the 1st circle on the left side of the background,
+                // and Secondary color paints the 2nd circle on the right side of the background:
+                if (index === 1) {
+                    circle.setAttribute('fill', custom_secondary_color || default_secondary_color);
+                } else circle.setAttribute('fill', custom_primary_color || default_primary_color);
+            });
+        }
+        // show card content when the background svg has loaded:
+        if (!is_content_shown) setIsContentShown(true);
+    };
+
+    useEffect(() => {
+        if (is_content_shown) updateBackground();
+    }, [dark, wallet_name, demo]);
+
+    const getCardText = () => {
+        if (size !== 'small' && balance) {
+            return (
+                <div>
+                    <Text as="div" type="extra-small" bold={false}>
+                        {demo ? 'Demo' : wallet_title} {fiat_wallets.every((f) => wallet_name !== f) && wallet_currency}{' '}
+                        wallet
+                    </Text>
+                    <Text as="div" type="paragraph-2" bold>
+                        {balance} {wallet_currency}
+                    </Text>
+                </div>
+            );
+        } else if (size !== 'small' && !balance) {
+            return (
+                <Text as="div" type="paragraph-2" bold={false}>
+                    {demo ? 'Demo' : wallet_title} wallet
+                </Text>
+            );
+        }
```

Comment:
@yashim-deriv for balance I agree but they both have to be executed only if size !== 'small'. If size is small, nothing should be returned. If I just return the second part without else if, it will be executed when size === small. However, of course, I can move the size check outside before we call the getCardText.. What do you think? 🙂

Reply:
Lets move the check out. Less nesting = easier to read :catjam: